src/bin/gosub-parser.rs

```diff
@@ -1,25 +1,19 @@
-use anyhow::Result;
+use anyhow::{bail, Result};
 use gosub_html5::parser::document::{Document, DocumentBuilder};
 use gosub_html5::parser::Html5Parser;
 use gosub_shared::bytes::{CharIterator, Confidence, Encoding};
 use std::fs;
-use std::process::exit;
-
-fn bail(message: &str) -> ! {
-    println!("{message}");
-    exit(1);
-}
 
 fn main() -> Result<()> {
     let url = std::env::args()
         .nth(1)
-        .unwrap_or_else(|| bail("Usage: gosub-parser <url>"));
+        .unwrap_or_else(|| bail!("Usage: gosub-parser <url>"));
 
     let html = if url.starts_with("http://") || url.starts_with("https://") {
         // Fetch the html from the url
         let response = ureq::get(&url).call()?;
         if response.status() != 200 {
-            bail(&format!(
+            bail!(&format!(
                 "Could not get url. Status code {}",
                 response.status()
             ));
```

Comment:
The bail macro supports formatting arguments so you don't have to explicitly use the format macro.
(See https://docs.rs/anyhow/latest/anyhow/macro.bail.html)
```suggestion
            bail!("Could not get url. Status code {}", response.status());
```